dbms/src/Storages/LiveView/StorageLiveView.cpp

```diff
@@ -177,11 +229,25 @@ void StorageLiveView::writeIntoLiveView(
         }
     }
 
-    auto parent_storage = context.getTable(live_view.getSelectDatabaseName(), live_view.getSelectTableName());
-    auto proxy_storage = std::make_shared<ProxyStorage>(parent_storage, std::move(from), QueryProcessingStage::WithMergeableState);
-    InterpreterSelectQuery select(live_view.getInnerQuery(), context, proxy_storage, QueryProcessingStage::Complete);
+    auto blocks_storage = StorageBlocks::createStorage(live_view.database_name, live_view.table_name, parent_storage->getColumns(), std::move(from), QueryProcessingStage::WithMergeableState);
+    InterpreterSelectQuery select(mergeable_query->clone(), context, blocks_storage, SelectQueryOptions(QueryProcessingStage::Complete));
     BlockInputStreamPtr data = std::make_shared<MaterializingBlockInputStream>(select.execute().in);
 
+    if (live_view.getInnerSubQuery())
```

Comment:
I think that we even don't need to execute inner subquery separately. If just use `live_view.getInnerQuery()` instead of `mergeable_query` above. Need to check it.

Reply:
Actually, we do need to execute inner subquery separately. This allows the mergeable blocks to always be generated based on the innermost subquery which is what we want.